index.js

```diff
@@ -70,48 +66,21 @@ const createWindow = () => {
   } = require('./utils/constants.js')
 
   ipcMain.on('start', (event, args) => {
-    downloadProcess.send({ path: basePath })
-    downloadProcess.on('message', (msg) => {
-      if (msg.start) {
-        // Return synchronously once the process has finished starting.
-        event.returnValue = null
-      } else {
-        event.sender.send(DOWNLOADED_RECV, msg)
-      }
+    const sender = new BulkSender((bulkMsg) => event.sender.send(DOWNLOADED_RECV, bulkMsg))
+    startQueue(app.getPath('userData'), 'queue.json', (msg) => sender.add(msg)).then((queue) => {
+      // Setup ipc handlers after queue started.
+      ipcMain.on(DOWNLOAD_CHAPTER_MSG, (event, args) => downloadChapter(event, args, queue))
+      ipcMain.on(DELETE_CHAPTER_MSG, (event, args) => returnAsync(args, deleteChapter(basePath, args), event, DELETE_CHAPTER_RECV))
+      ipcMain.on(DELETE_MANGA_MSG, (event, args) => deleteManga(basePath, args))
+
+      // Signal to the renderer that the queue has finished starting.
+      event.returnValue = null
     })
   })
-  ipcMain.on(DOWNLOAD_CHAPTER_MSG, (event, args) => downloadChapter(event, args, downloadProcess))
-  ipcMain.on(DELETE_CHAPTER_MSG, (event, args) => returnAsync(args, deleteChapter(basePath, args), event, DELETE_CHAPTER_RECV))
-  ipcMain.on(DELETE_MANGA_MSG, (event, args) => deleteManga(basePath, args))
 }
 
-if (cluster.isMaster) {
-  downloadProcess = cluster.fork()
-
-  app.on('ready', createWindow)
-
-  app.on('window-all-closed', function () {
-    // So Mac applications don't close when windows closed.
-    if (process.platform !== 'darwin') {
-      app.quit()
-    }
-  })
-
-  app.on('activate', function () {
-    // So Mac applications can be recreated when the dock icon is clicked.
-    if (mainWindow === null) {
-      createWindow()
-    }
-  })
-} else {
-  // Download process code.
-  process.on('message', (msg) => {
-    if (msg.path) {
-      const { downloadProcessHandler } = require('./utils/process.js')
-      downloadProcessHandler(msg.path, 'queue.json')
-    }
-  })
-}
+app.on('ready', createWindow)
+app.on('window-all-closed', () => app.quit())
```

Comment:
Good. The app will never quit unless if you explicitly quit it. Strange Electron.